### PR TITLE
Steam-deploying

### PR DESCRIPTION
**Pull request by mohammedbabelly20**

### CODE REVIEW

File: content/yaml-publishing/steam.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+{{</notebox>}}
+
+## Prerequisites
+
+- A Unity **Plus** or **Pro** license. Your license is used to activate Unity on the Codemagic build server so the project can be built. The license is returned automatically once the workflow completes.
+- A Steam Partner account is required to publish to Steam.
+
+## Getting Started
+
+Steamcmd is the tool used to upload builds to Steam. Steamcmd requires logging into Steam and will normally require a Steam Guard code be entered.
```
Comment (VeArnold) [inline content/yaml-publishing/steam.md:20]: in to

Comment (VeArnold) [inline content/yaml-publishing/steam.md:20]: SteamCMD?

File: content/yaml-publishing/steam.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+To solve this problem, there are two options.
+
+1. Disable Steam Guard for the account doing the Steam upload.  This is not recommended, as it makes the Steam account less secure.
+
+2. Use sentry files that are generated after logging in successfully to Steam on the build machine. When these sentry files are present on the build machine, a Steam Guard code is not required.
+So we are going to save the sentry files as secure environment variables and then place them at the correct path when the build starts.
+
+### Obtain the sentry files:
+First you need to install the steamcmd.
```
Comment (VeArnold) [inline content/yaml-publishing/steam.md:29]: First, you need to install SteamCMD.

File: content/yaml-publishing/steam.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+You can now see the **ssfn** file in `~/Steam/ssfn*******************` and the **config.vdf** file at `~/Steam/config/config.vdf`.
+
+{{<notebox>}}
+Warning: Keep the sentry files private; don't check them into public source control.
+{{</notebox>}}
+
+- Save the ssfn file name, ssfn file itself, and the config file to the respective environment variables in the **Environment variables** section in Codemagic UI, so they can be used in subsequent builds. Click **Secure** to encrypt the values. Note that binary files (i.e. ssfn, config.vdf) have to be [`base64 encoded`](../variables/environment-variable-groups/#storing-sensitive-valuesfiles) locally before they can be saved to environment variables and decoded during the build.
+
+{{<notebox>}}
+Warning: When you are base64 encoding the files, make sure to remove the new lines `\n` of the encoded value, before saving it to Codemagic.
```
Comment (VeArnold) [inline content/yaml-publishing/steam.md:62]: value, before (unnecessary comma)

File: content/yaml-publishing/steam.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+These are standard VDF files required for uploading a build to Steam and require your application's AppID, DepotID and branch name for deployment.
+
+And then use the script to publish your app to steam:
+```yaml
+      - name: Upload Build to Steam
+        script: | 
+          ~/Steam/steamcmd.sh +login $STEAM_USERNAME $STEAM_PASSWORD +run_app_build ~/clone/steam/app_build.vdf +quit
+```
+
+See the sample project in [here](https://github.com/codemagic-ci-cd/codemagic-sample-projects/tree/main/unity/unity-deploy-steam).
```
Comment (VeArnold) [inline content/yaml-publishing/steam.md:121]: remove "in"

**Review by VeArnold — CHANGES_REQUESTED** [review]

File: content/yaml-publishing/steam.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+{{</notebox>}}
+
+## Prerequisites
+
+- A Unity **Plus** or **Pro** license. Your license is used to activate Unity on the Codemagic build server so the project can be built. The license is returned automatically once the workflow completes.
+- A Steam Partner account is required to publish to Steam.
+
+## Getting Started
+
+SteamCMD is the tool used to upload builds to Steam. SteamCMD requires logging into Steam and will normally require a Steam Guard code be entered.
```
Comment (VeArnold) [inline content/yaml-publishing/steam.md:20]:
```suggestion
SteamCMD is a tool used to upload builds to Steam. SteamCMD requires logging in to Steam and typically requires entering a Steam Guard code.
```

File: content/yaml-publishing/steam.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+To solve this problem, there are two options.
+
+1. Disable Steam Guard for the account doing the Steam upload.  This is not recommended, as it makes the Steam account less secure.
+
+2. Use sentry files that are generated after logging in successfully to Steam on the build machine. When these sentry files are present on the build machine, a Steam Guard code is not required.
+So we are going to save the sentry files as secure environment variables and then place them at the correct path when the build starts.
```
Comment (VeArnold) [inline content/yaml-publishing/steam.md:26]: Thus, we will save the sentry files as secure environment variables and place them at the correct path when the build starts.

File: content/yaml-publishing/steam.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+You can now see the **ssfn** file in `~/Steam/ssfn*******************` and the **config.vdf** file at `~/Steam/config/config.vdf`.
+
+{{<notebox>}}
+Warning: Keep the sentry files private; don't check them into public source control.
+{{</notebox>}}
+
+- Save the ssfn file name, ssfn file itself, and the config file to the respective environment variables in the **Environment variables** section in Codemagic UI, so they can be used in subsequent builds. Click **Secure** to encrypt the values. Note that binary files (i.e. ssfn, config.vdf) have to be [`base64 encoded`](../variables/environment-variable-groups/#storing-sensitive-valuesfiles) locally before they can be saved to environment variables and decoded during the build.
+
+{{<notebox>}}
+Warning: When you are base64 encoding the files, make sure to remove the new lines `\n` of the encoded value before saving it to Codemagic.
```
Comment (VeArnold) [inline content/yaml-publishing/steam.md:62]:
```suggestion
Warning: When base64 encoding the files, make sure to remove the new lines `\n` of the encoded value before saving it to Codemagic. 
```

File: content/yaml-publishing/steam.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+{{< /tabpane >}}
+
+### Upload to Steam
+To configure the upload to Steam, edit the following two files in the demo project:
+```
+   steam/app_build.vdf
+   steam/depot_build.vdf
+```
+These are standard VDF files required for uploading a build to Steam and require your application's AppID, DepotID and branch name for deployment.
```
Comment (VeArnold) [inline content/yaml-publishing/steam.md:112]:
```suggestion
These are standard VDF files required for uploading a build to Steam and require your application's AppID, DepotID, and branch name for deployment.
```

**Review by VeArnold — CHANGES_REQUESTED** [review]

**Comment by VeArnold** [comment]
Can merge after looking over the suggestions. Other than minor changes, LGTM.

**Comment by codemagic-mohammed** [comment]
> Can merge after looking over the suggestions. Other than minor changes, LGTM.

Ok, I applied the suggested changes, you can accept the PR now so I can merge it.

**Review by VeArnold — APPROVED** [review]
LGTM

**Pull request merged**